src/nvim/api/buffer.c

```diff
@@ -395,10 +395,10 @@ void nvim_buf_set_lines(uint64_t channel_id,
     mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
   }
 
-  changed_lines((linenr_T)start, 0, (linenr_T)end, extra);
+  changed_lines((linenr_T)start, 0, (linenr_T)end, (long)extra);
```

Comment:
Can't we change the type of `extra` to `linenr_T` or `long`? Guranteed value range of `ssize_t` is `[-1, SSIZE_MAX]`, but the value saved in `extra` could be smaller than `-1`, see line 346.